nomenclature/codes_models.py

```diff
@@ -0,0 +1,74 @@
+from re import compile, match
+from typing import Union, List, Dict
+from pydantic import BaseModel, validator
+
+
+TAG_PATTERN = compile("<.*>$")
+
+
+class Code(BaseModel):
+    """A simple class for a mapping of a "code" to its attributes"""
+
+    name: str
+    attributes: Dict[str, Union[str, dict, None]]
```

Comment:
You could further specify the structure of the dictionary inside the `Union`

Reply:
Done